content/development/regex_assembly.md

```diff
@@ -1,84 +1,15 @@
 
-## Prerequisites
+## Specification Format
 
-* A Perl environment and Perl version >= 5.10
-* A Python environment with Python version >= 3
-
-## Setup
-
-1. Initialize the Git submodule that contains the Regexp::Assemble Perl module by running:
-
-    ```bash
-    git submodule update --init util/regexp-assemble/lib/lib
-    ```
-
-1. Install some Python dependencies. With `virtualenv`, installing them would look like this:
-
-    ```bash
-    virtualenv -p 3 venv
-    source venv/bin/activate
-    pip install -r util/regexp-assemble/requirements.txt
-    ```
-
-1. The `regexp-assemble.py` script should now be usable. To test, try running something like the following:
-
-    ```bash
-    printf "(?:homer)? simpson\n(?:lisa)? simpson" | util/regexp-assemble/regexp-assemble.py generate -
-    ```
-
-    The output should be:
-
-    ```bash
-    (?:(?:homer)?|(?:lisa)?) simpson
-    ```
-
-## Example Use
-
-To generate a reduced expression from a list of expressions, simply pass the corresponding CRS rule ID to the script or pipe the contents to it:
-
-```bash
-util/regexp-assemble/regexp-assemble.py generate 942170
-# or
-cat util/regexp-assemble/data/942170.data | util/regexp-assemble/regexp-assemble.py generate -
-```
-
-It is also possible to compare generated expressions to the current expressions in the rule files, like so:
-
-```bash
-util/regexp-assemble/regexp-assemble.py compare 942170
-```
-
-Even better, rule files can be updated directly:
-
-```bash
-util/regexp-assemble/regexp-assemble.py update 942170
-# or update all
-util/regexp-assemble/regexp-assemble.py update --all
-```
-
-Read the built-in help text for the full documentation:
-
-```bash
-util/regexp-assemble/regexp-assemble.py --help
-```
-
-## Adjusting the Logging Level
-
-The level of logging can be adjusted with the `--loglevel` option. Accepted values are `DEBUG`, `INFO`, `WARNING`, `ERROR`, and `CRITICAL`.
-
-Not many things are currently written to the log, so it may seem that changing the logging level doesn't have any effect. This is especially true since the processing modules depend on the rule that the script processes and the command that is being run (e.g., `generate` usually produces no log output since stdout is used to print the resulting regular expression). The `cmdline` processor, for example, includes debug log output, but the `assemble` processor currently doesn't.
-
-## Data File Format
-
-The data files (`.data` suffix, under `util/regexp-assemble/data`) contain one regular expression per line. These files are meant to be processed by [regexp-assemble.py](regexp-assemble.py).
+The files containing regular expression specifications (`.data` suffix, under `data`) contain one regular expression per line. These files are meant to be processed by the [crs-toolchain]({{< ref "crs_toolchain" >}}).
```

Comment:
```suggestion
The files containing regular expression specifications (`.ra` suffix, under `data`) contain one regular expression per line. These files are meant to be processed by the [crs-toolchain]({{< ref "crs_toolchain" >}}).
```

Reply:
@RedXanadu suggested renaming the `data` directory to something better. Maybe `regex-assembly`? We could make the change now, which will probably lead to less disruption later... What do you think?

Reply:
Or something like `regex-src`? Anyway, it's a bit of bikeshedding. But I think `regexp-assembly` is the most clear (`data` could mean everything and it's also used for pmFromFile) and I also think the `.ra` extension is a GREAT idea.

Reply:
Thanks @lifeforms, I agree.